core/web/resolver/ocr.go

```diff
@@ -51,32 +53,40 @@ type CreateOCRKeyBundlePayloadResolver struct {
 }
 
 func NewCreateOCRKeyBundlePayloadResolver(key ocrkey.KeyV2) *CreateOCRKeyBundlePayloadResolver {
-	return &CreateOCRKeyBundlePayloadResolver{key}
+	return &CreateOCRKeyBundlePayloadResolver{key: key}
 }
 
 func (r *CreateOCRKeyBundlePayloadResolver) Bundle() OCRKeyBundleResolver {
-	return OCRKeyBundleResolver{r.key}
+	return OCRKeyBundleResolver{key: r.key}
 }
 
 type DeleteOCRKeyBundleSuccessResolver struct {
 	key ocrkey.KeyV2
 }
 
 func NewDeleteOCRKeyBundleSuccessResolver(key ocrkey.KeyV2) *DeleteOCRKeyBundleSuccessResolver {
-	return &DeleteOCRKeyBundleSuccessResolver{key}
+	return &DeleteOCRKeyBundleSuccessResolver{key: key}
 }
 
 func (r *DeleteOCRKeyBundleSuccessResolver) Bundle() OCRKeyBundleResolver {
-	return OCRKeyBundleResolver{r.key}
+	return OCRKeyBundleResolver{key: r.key}
 }
 
 type DeleteOCRKeyBundlePayloadResolver struct {
 	key ocrkey.KeyV2
-	err error
+	NotFoundErrorUnionType
 }
 
 func NewDeleteOCRKeyBundlePayloadResolver(key ocrkey.KeyV2, err error) *DeleteOCRKeyBundlePayloadResolver {
-	return &DeleteOCRKeyBundlePayloadResolver{key, err}
+	var e NotFoundErrorUnionType
+
+	if err != nil {
```

Comment:
Do we need the `if err != nil` check here? Why not just always set it?

Reply:
We need it cause this part of the code is calling `err.Error()` and that would fail if `NewDeleteOCRKeyBundlePayloadResolver` is called without an error, which it happens.

Reply:
Ah absolutely right!